app/javascript/shared/components/MarkdownEditor.res

```diff
@@ -121,6 +121,22 @@ let modeIcon = (desiredMode, currentMode) => {
   <FaIcon classes={"fa-fw " ++ icon} />
 }
 
+let modeLabel = (desiredMode, currentMode) => {
+  let label = switch (desiredMode, currentMode) {
+  | (#Preview, Windowed(#Editor) | Fullscreen(#Editor) | Fullscreen(#Split)) => "Preview"
+  | (#Preview, Windowed(#Preview) | Fullscreen(#Preview)) => "Edit"
+  | (
+      #Split,
+      Windowed(_) | Fullscreen(#Editor) | Fullscreen(#Preview),
+    ) => "View editor and preview in split view"
+  | (#Split, Fullscreen(#Split)) => "Close split view"
+  | (#Fullscreen, Windowed(_)) => "Full screen"
+  | (#Fullscreen, Fullscreen(_)) => "Exit full screen"
+  }
+
+  label
+}
```

Comment:
The `label` binding is probably not required here.

```suggestion
let modeLabel = (desiredMode, currentMode) =>
  switch (desiredMode, currentMode) {
  | (#Preview, Windowed(#Editor) | Fullscreen(#Editor) | Fullscreen(#Split)) => "Preview"
  | (#Preview, Windowed(#Preview) | Fullscreen(#Preview)) => "Edit"
  | (
      #Split,
      Windowed(_) | Fullscreen(#Editor) | Fullscreen(#Preview),
    ) => "View editor and preview in split view"
  | (#Split, Fullscreen(#Split)) => "Close split view"
  | (#Fullscreen, Windowed(_)) => "Full screen"
  | (#Fullscreen, Fullscreen(_)) => "Exit full screen"
  }
```

Reply:
This is needed for the accessibility of buttons in the toolbar. The reason for having these bindings is;



1. We are using **_Icon-only buttons_** in the toolbar of the Markdown editor, hence need labels for them.
2. The same button has different usage in different modes. Eg. once we click on the preview button, the same button turns into an edit button.

Reply:
@irajsuhail I wasn't saying that we don't need the _label_. I was saying that we don't need the _binding_ - the named binding called `label` in this method. Take a look at my suggestion in this comment. 😁

Reply:
Sorry, I misunderstood it. 😅